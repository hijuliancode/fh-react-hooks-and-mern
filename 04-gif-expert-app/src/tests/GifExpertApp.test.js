import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import GifExpertAppComponent from '../GifExpertApp'


describe('Test in GifExpertApp.js', () => {
  let wrapper;

  test('should render <GifExpertApp />', () => {
    wrapper = shallow( <GifExpertAppComponent /> )
    expect( wrapper ).toMatchSnapshot()
  })

  test('should render a categories list', () => {
    const categories = [ { id: 'DragonBall', name: 'Dragon Ball' }, { id: 'IronMan', name: 'Iron Man' } ]
    const wrapper = shallow( <GifExpertAppComponent defaultCategories={ categories } /> )
    
    expect( wrapper ).toMatchSnapshot()
    expect( wrapper.find('GifGrid').length ).toBe( categories.length )

  })

})