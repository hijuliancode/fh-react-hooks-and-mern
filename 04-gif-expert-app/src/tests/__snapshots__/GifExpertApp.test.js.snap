// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in GifExpertApp.js should render <GifExpertApp /> 1`] = `
<div
  className="container"
>
  <h2
    className="is-size-5-mobile is-size-4 has-text-centered mt-4"
  >
    Gif Expert App
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
</div>
`;

exports[`Test in GifExpertApp.js should render a categories list 1`] = `
<div
  className="container"
>
  <h2
    className="is-size-5-mobile is-size-4 has-text-centered mt-4"
  >
    Gif Expert App
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <GifGrid
    category="Dragon Ball"
    key="DragonBall"
  />
  <GifGrid
    category="Iron Man"
    key="IronMan"
  />
</div>
`;
